atlasdb-backups/src/main/java/com/palantir/atlasdb/backup/AtlasBackupTask.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.backup;
+
+import com.palantir.atlasdb.backup.api.AtlasBackupServiceBlocking;
+import com.palantir.atlasdb.timelock.api.BackupToken;
+import com.palantir.atlasdb.timelock.api.CompleteBackupRequest;
+import com.palantir.atlasdb.timelock.api.CompleteBackupResponse;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.atlasdb.timelock.api.PrepareBackupRequest;
+import com.palantir.atlasdb.timelock.api.PrepareBackupResponse;
+import com.palantir.conjure.java.api.config.service.ServicesConfigBlock;
+import com.palantir.dialogue.clients.DialogueClients;
+import com.palantir.dialogue.clients.DialogueClients.ReloadingFactory;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.tokens.auth.AuthHeader;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class AtlasBackupTask {
+    private final AtlasBackupServiceBlocking atlasBackupServiceBlocking;
+    private final BackupTokenPersister backupTokenPersister;
+
+    private AtlasBackupTask(AtlasBackupServiceBlocking atlasBackupServiceBlocking) {
+        this.atlasBackupServiceBlocking = atlasBackupServiceBlocking;
+        this.backupTokenPersister = new InMemoryBackupTokenPersister();
```

Comment:
this should work if the parent tasks are run from the same node without a restart - to make this survive restarts, we'd want to write to disk (as the internal product currently does)